pkg/configs/api/api.go

```diff
@@ -111,12 +125,28 @@ func (a *API) setConfig(w http.ResponseWriter, r *http.Request) {
 	logger := util.WithContext(r.Context(), util.Logger)
 
 	var cfg configs.Config
-	if err := json.NewDecoder(r.Body).Decode(&cfg); err != nil {
-		// XXX: Untested
-		level.Error(logger).Log("msg", "error decoding json body", "err", err)
+	switch ParseConfigFormat(r.Header.Get("Content-Type"), FormatJSON) {
+	case FormatJSON:
+		if err := json.NewDecoder(r.Body).Decode(&cfg); err != nil {
+			// XXX: Untested
+			level.Error(logger).Log("msg", "error decoding json body", "err", err)
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+	case FormatYAML:
+		if err := yaml.NewDecoder(r.Body).Decode(&cfg); err != nil {
+			// XXX: Untested
+			level.Error(logger).Log("msg", "error decoding yaml body", "err", err)
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+	default:
+		// should not be here
+		level.Error(logger).Log("msg", "unknown Format")
```

Comment:
No need to log this error (actually, not an error, just invalid input from the client).

Reply:
actually, this block is unreachable because parseConfigFormat has a default value.